### PR TITLE
Add ``--show-capabilities``

### PR DESCRIPTION
**Pull request by Asnelchristian**
Closes https://github.com/coala-analyzer/coala/issues/2431

### CODE REVIEW

File: coalib/coala.py [hunk 1]
```diff
@@ -45,13 +46,23 @@ def main():
...
                     global_bears, args.filter_by_language)
-
             show_bears(local_bears,
                        global_bears,
                        args.show_description or args.show_details,
                        args.show_details,
                        console_printer)
             return 0
+        else:
+            if args.show_capabilites:
```
Comment (Adrianzatreanu) [inline coalib/coala.py]: Show_capabilities** typo here

Comment (Adrianzatreanu) [inline coalib/coala.py]: Ah now i see, you named the function badly :d

File: coalib/parsing/DefaultArgParser.py [hunk 2]
```diff
@@ -170,6 +170,10 @@ def default_arg_parser(formatter_class=None):
             help="filters `--show-bears` by the given languages")
 
         outputs_group.add_argument(
+           '-x', '--show-capabilites', nargs='+', metavar='LANG',
```
Comment (sils1297) [inline coalib/parsing/DefaultArgParser.py]: why x? Nothing else left? Any upper case letter that relates to it somehow more than x?

File: coalib/output/ConsoleInteraction.py [hunk 3]
```diff
@@ -751,3 +753,36 @@ def show_bears(local_bears,
...
+        Object to print messages on the console.
+    """
+    if not language_bears_capabilities:
+        console_printer.print("No language was passed")
+    else:
+        for language, capabilities in language_bears_capabilities.items():
+            if capabilities[0]:
+                console_printer.print('coala can do the following for ', end='')
+                console_printer.print(language, color="blue")
+                console_printer.print("\tCan detect: ", end='')
```
Comment (sils1297) [inline coalib/output/ConsoleInteraction.py]: can you look at the show_bear method? This looks like it should be added there. Will be less code as well.

File: coalib/parsing/DefaultArgParser.py [hunk 4]
```diff
@@ -170,6 +170,10 @@ def default_arg_parser(formatter_class=None):
...
+           help="show languages' bears capabilities")
```
Comment (sils1297) [inline coalib/parsing/DefaultArgParser.py]: I don't fully get what this does, could you explain? I feel like the description needs to be  improved.

File: coalib/parsing/DefaultArgParser.py (hunk 2 above)
Comment (Adrianzatreanu) [inline coalib/parsing/DefaultArgParser.py]: Why -X? It is not suggestive at all.

File: coalib/output/ConsoleInteraction.py (hunk 3 above)
Comment (Asnelchristian) [inline coalib/output/ConsoleInteraction.py]: okay :)

File: coalib/collecting/Collectors.py [hunk 5]
```diff
@@ -194,6 +194,35 @@ def filter_section_bears_by_languages(bears, languages):
...
+    language_bears_capabilities = {language: (
+        set(), set()) for language in languages}
+    for section in bears.keys():
+        for bear in bears[section]:
+            bear_language = (
+                {language.lower() for language in bear.LANGUAGES} & languages)
+            if bear_language:
+                language = bear_language.pop()
+                capabilities = language_bears_capabilities[language]
+                language_bears_capacities[language] = (
```
Comment (sils1297) [inline coalib/collecting/Collectors.py]: missing coverage! That can't be merged.

File: coalib/collecting/Collectors.py [hunk 6]
```diff
@@ -194,6 +194,35 @@ def filter_section_bears_by_languages(bears, languages):
     return new_bears
 
 
+def get_bears_capabilites_filter_by_languages(bears, languages):
```
Comment (sils1297) [inline coalib/collecting/Collectors.py]: how about `filter_capabilities_by_lang`?

File: coalib/collecting/Collectors.py [hunk 7]
```diff
@@ -194,6 +194,35 @@ def filter_section_bears_by_languages(bears, languages):
...
+    """
+    Gets languages' bears capabilities.
```
Comment (sils1297) [inline coalib/collecting/Collectors.py]: oh then it doesn't filter a thing, the name is misleading IMO

File: coalib/output/ConsoleInteraction.py [hunk 8]
```diff
@@ -751,3 +753,36 @@ def show_bears(local_bears,
...
+    :param language_bears_capabilities:
+        Dictionary with languages as keys and their bears capabilities as
+        values. The capabilities are stored in a tuple of two element where the
+        first one represent what the bears can detect, and the second one what
+        they can fix.
+    :param console_printer:
+        Object to print messages on the console.
+    """
+    if not language_bears_capabilities:
+        console_printer.print("No language was passed")
```
Comment (sils1297) [inline coalib/output/ConsoleInteraction.py]: or maybe there are no bears

File: coalib/output/ConsoleInteraction.py (hunk 3 above)
Comment (sils1297) [inline coalib/output/ConsoleInteraction.py]: why are you using tabs? We're not using them in other places, rather put spaces.

File: coalib/output/ConsoleInteraction.py [hunk 9]
```diff
@@ -751,3 +753,36 @@ def show_bears(local_bears,
...
+            if capabilities[0]:
+                console_printer.print('coala can do the following for ', end='')
+                console_printer.print(language, color="blue")
+                console_printer.print("\tCan detect: ", end='')
+                console_printer.print(
+                    ', '.join(capabilities[0]), color=CAN_DETECT_COLOR)
+                if capabilities[1]:
+                    console_printer.print("\tCan fix   : ", end='')
+                    console_printer.print(
+                        ', '.join(capabilities[1]), color=CAN_FIX_COLOR)
```
Comment (sils1297) [inline coalib/output/ConsoleInteraction.py]: I think having different colours here makes the stuff hard to read

File: coalib/output/ConsoleInteraction.py [hunk 10]
```diff
@@ -751,3 +753,36 @@ def show_bears(local_bears,
...
+            else:
+                console_printer.print('coala does not support ', color='red',
+                                      end='')
+                console_printer.print(language, color='blue')
```
Comment (sils1297) [inline coalib/output/ConsoleInteraction.py]: uhm, you should mention the generic bears here

Comment (sils1297) [inline coalib/output/ConsoleInteraction.py]: or maybe the user did just a typo

**Comment by sils1297** [comment]
https://github.com/coala-analyzer/coala/pull/2437/commits/cf16f54719014a13a69d65b4d84fbd408b9ef8af needs work

**Comment by sils1297** [comment]
Ping me please when you need a review :)

**Comment by Adrianzatreanu** [comment]
moved to https://github.com/coala-analyzer/coala/pull/2443

**Pull request closed without merging**